accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,305 @@
+
+This attestation is using the tool `snpguest` which is provided in SUSE Linux Enterprise Server 15 SP6 and later.
+It is made for AMD EPYC CPUs of type Milan and Genoa.
+The host and the hypervisor must support Confidential VMs to perform a successful attestation.
+
+The tool and the possible attestation workflows are described in the [README](https://github.com/virtee/snpguest/blob/main/README.md).
+
+The proposed workflow for attestation using SUSE Manager is:
+
+![SNP Guest Attestation](images/coco-attestation/CoCoAttestation.png)
+
+- Apply a salt state on the VM
+  - create the entry on the report table with status `pending`
+  - assign to a systemid and an execution date (create date of the report)
+  - add the nonce to the entry in `in_data` and set it as pillar data
+  - request the attestation report
+- Store the state.apply result data into the database
+  - Store report data into `out_data`
+  - check the expected validation results and add all the entries into the result table with status `pending`
+- Run a Container which does the attestation
+  - search the DB for pending attestation
+  - try to lock row (the job)
+  - request the vcek certificate (require internet access of the container)
+  - verify the certificates and the signatures
+  - verify the nonce is in the report
+  - write the result into the DB
+  - set the status of the result entry to `succeeded` or `failed`
+
+Optionally the VCEK could be read from the extended memory if this is supported by the environment.
+In this case the VCEK is send together with the attestation report and stored in the database.
+
+Additionally it should be tested if secure boot is enabled. This should also fill in the result entry and set the status.
+
+A trigger is setting the status of the Report Table. When all required check were successful, the status is set to `succeeded`.
+If one required check is set to failed, the overall status is set to `failed` as well.
+In case there are still pending checks, the status stay in `pending`.
+
+
+### The Attestation Container for SEV-SNP attestation
+
+An attestation container is responsible for 1 result type.
+A container can be enhanced and act for more result types if no other tooling is needed and to build another container would be overkill.
+
+Therefor the Container should contain the just the `snpguest` tool and the CA Certificates for the different Processor Models (ARK and ASK for Milan and Genoa).
+In case the VCEK is available, the verification can directly start and the container only needs access to the database.
+If the VCEK is not available, the container needs to fetch it from the AMD Key Distribution Center.
+This require internet access for the attestation container.
+
+The program which start the verification should regular check the DB for pending attestations.
+It should fork a process for a specific report result.
+It can use DB locking mechanisms to exclusively work on a specific report.
```

Comment:
Should it lock the report or the results it can proccess?

Reply:
yes, the text needs to be adapted